### PR TITLE
Update snappy to newer version

### PR DESCRIPTION
**Pull request by maarten-k**
### Description

see issue #666 

please be gentle :This is my first pull request and not a full time  java developer ;) 

### Checklist

- [x] Code compiles correctly
- [ ] All tests passing (tests about testDate failed, but where untouched)

### CODE REVIEW

**Comment by lbergelson** [comment]
@maarten-k Thanks for the pull request!  Welcome to the project!  Sorry I haven't gotten time to do a proper review yet.  

One note, I think that we should upgrade to a regular release if we're going to do an upgrade instead of a release candidate.  It looks like 1.1.2.6 is the last regular release.  (We can upgrade again easily once 1.1.4 is officially released.  It sounds like they don't have all of the native libraries available yet in the [1.1.4-M3 preview](https://github.com/xerial/snappy-java/blob/master/Milestone.md#snappy-java-114-will-be-available-early-2017).

**Comment by maarten-k** [comment]
Sound like a good idea. I downgraded to the stable version

File: src/main/java/htsjdk/samtools/util/SnappyLoader.java [hunk 1]
```diff
@@ -64,42 +60,13 @@ public SnappyLoader(final boolean verbose) {
...
-            } catch (Error e) {
-                if (e.getClass().equals(snappyErrorClass)) {
-                    if (verbose) System.err.println("Snappy dll failed to load: " + e.getMessage());
-                    tmpSnappyAvailable = false;
-                } else {
-                    throw e;
-                }
-            }
-            SnappyAvailable = tmpSnappyAvailable;
+            SnappyAvailable = true;
```
Comment (yfarjoun) [inline src/main/java/htsjdk/samtools/util/SnappyLoader.java]: codeCov shows that this line is never executed in tests. I suspect that this is not due to lack of tests using gzip. 

The constructor objects are always null and so the if statement is always false...please fix.

**Comment by tfenne** [comment]
I think (as @yfarjoun points out) this PR doesn't work - it deletes much of the code to detect whether snappy is actually available or not.  I've prepared an alternative PR, #872, and would suggest closing this one.

**Comment by yfarjoun** [comment]
Thanks for submitting this PR @maarten-k we are very happy for the contribution. it seems that the best way forward is to focus on @tfenne 's PR #872. So I'll close this one.

**Comment by maarten-k** [comment]
@yfarjoun great to see someone picked this up!

**Pull request closed without merging**